crates/paas/src/error.rs

```diff
@@ -0,0 +1,94 @@
+//! Error types for PaaS
+
+use thiserror::Error;
+
+/// Result type for PaaS operations
+pub type ProverServiceResult<T> = Result<T, ProverServiceError>;
+
+/// PaaS error types
+#[derive(Error, Debug)]
+pub enum ProverServiceError {
+    /// Task not found
+    #[error("Task not found: {0}")]
+    TaskNotFound(String),
+
+    /// Transient failure that should be retried
+    #[error("Transient failure: {0}")]
+    TransientFailure(String),
+
+    /// Permanent failure that should not be retried
+    #[error("Permanent failure: {0}")]
+    PermanentFailure(String),
+
+    /// Invalid state transition
+    #[error("Invalid state transition from {from:?} to {to:?}")]
+    InvalidTransition {
+        from: crate::task::TaskStatus,
+        to: crate::task::TaskStatus,
+    },
+
+    /// Worker pool error
+    #[error("Worker pool error: {0}")]
+    WorkerPool(String),
+
+    /// Configuration error
+    #[error("Configuration error: {0}")]
+    Config(String),
+
+    /// Internal error
+    #[error("Internal error: {0}")]
+    Internal(#[from] anyhow::Error),
```

Comment:
Redundant to use `error` in error messages.

Also is there some slightly more clever `From`/`Into` impls we can do in the `anyhow::Error` case to keep the context propagation linked together nicer?